### DISCUSSION TITLE
SSR example always fetch client side

### DISCUSSION THREAD

**Discussion by lu-zen** [tiny-frontend/tiny-client#10] [Q&A] [answered]
Hey.

On nextjs example I noticed the ssr has this:
`  const ExampleTinyFrontend =
    TinyFrontendServerStorage.Component ?? ExampleTinyFrontendClient;`

It always fetch and import the component client side, but it's supposed to fetch and render server side.
What's the serverstorage btw?

**Comment by adrienbaron** [comment, marked as answer]
Hi @lu-zen !

You can find a detailed explanation about how this example works [here](https://tiny-frontend.github.io/guide/architecture.html#next-js).
Basically this `TinyFrontendServerStorage.Component` get set by the server when it's loading so it's only defined server side. This means that server side `ExampleTinyFrontendClient` don't get called (and therefore loaded), and client side as `TinyFrontendServerStorage.Component` is not defined, `ExampleTinyFrontendClient` gets used instead :).

Hope this makes sense!
